crates/bevy_animation/src/lib.rs

```diff
@@ -113,33 +110,87 @@ impl Default for AnimationPlayer {
+    current_weight: f32,
+    /// How much to decrease `current_weight` per second
+    weight_decline_per_sec: f32,
+    /// The animation that is being faded out
+    animation: PlayingAnimation,
+}
+
+/// Animation controls
+#[derive(Component, Default, Reflect)]
+#[reflect(Component)]
+pub struct AnimationPlayer {
+    paused: bool,
+
+    animation: PlayingAnimation,
+
+    // List of previous animations we're currently transitioning away from.
+    // Usually this is empty, when transitioning between animations, there is
+    // one entry. When another animation transition happens while a transition
+    // is still ongoing, then there can be more than one entry.
+    // Once a transition is finished, it will be automatically removed from the list
+    #[reflect(ignore)]
+    transitions: Vec<AnimationTransition>,
+}
+
 impl AnimationPlayer {
     /// Start playing an animation, resetting state of the player
-    pub fn start(&mut self, handle: Handle<AnimationClip>) -> &mut Self {
-        *self = Self {
+    /// If `transition_duration` is set, this will use a linear blending
+    /// between the previous and the new animation to make a smooth transition
+    pub fn start(
+        &mut self,
+        handle: Handle<AnimationClip>,
+        transition_duration: Option<Duration>,
+    ) -> &mut Self {
+        let mut animation = PlayingAnimation {
             animation_clip: handle,
             ..Default::default()
         };
+        std::mem::swap(&mut animation, &mut self.animation);
+        if let Some(transition_duration) = transition_duration {
+            // Add the current transition. If other transitions are still ongoing,
+            // this will keep those transitions running and cause a transition between
+            // the output of that previous transition to the new animation.
+            self.transitions.push(AnimationTransition {
+                current_weight: 1.0,
+                weight_decline_per_sec: 1.0 / transition_duration.as_secs_f32(),
+                animation,
+            });
+        } else {
+            // We want a hard transition.
+            // In case any previous transitions are still playing, stop them
+            self.transitions.clear();
+        }
         self
     }
 
     /// Start playing an animation, resetting state of the player, unless the requested animation is already playing.
-    pub fn play(&mut self, handle: Handle<AnimationClip>) -> &mut Self {
-        if self.animation_clip != handle || self.is_paused() {
-            self.start(handle);
+    /// If `transition_duration` is set, this will use a linear blending
+    /// between the previous and the new animation to make a smooth transition
+    pub fn play(
+        &mut self,
+        handle: Handle<AnimationClip>,
+        transition_duration: Option<Duration>,
+    ) -> &mut Self {
+        if self.animation.animation_clip != handle || self.is_paused() {
+            self.start(handle, transition_duration);
```

Comment:
`self.start()` will reset the player, i.e. it will run the new animation even if the previous one was paused. So it should already resume a paused animation. Or did I misunderstand you?

Reply:
Ah, that makes sense.